conda_smithy/configure_feedstock.py

```diff
@@ -878,6 +881,39 @@ def migrate_combined_spec(combined_spec, forge_dir, config, forge_config):
     return combined_spec
 
 
+def reduce_variants(recipe_path, config, input_variants):
+    """Subset of render_recipe to compute reduced variant matrix
+
+    large numbers of unused variants greatly increase render time
+    """
+
+    # from render_recipe
+    with conda_build.render.open_recipe(recipe_path) as recipe:
+        metadata = MetaData(str(recipe), config=config)
+
+    # from distribute_variants
+    # explode variants dict to list of variants
+    variants = initial_variants = conda_build.variants.get_package_variants(
+        metadata, variants=input_variants
+    )
+    logger.debug(f"Starting with {len(initial_variants)} variants")
+    metadata.config.variant = variants[0]
+    metadata.config.variants = variants
+    all_used = metadata.get_used_vars(force_global=True)
```

Comment:
What does `force_global=True` do here? That seems like it is worth a comment.

Reply:
Good call. force_global [looks at the whole recipe](https://github.com/conda/conda-build/blob/d1ba529d606ecc886331d1672f49b83c5c3e81bb/conda_build/metadata.py#L2922-L2928) for used variables, rather than parsing outputs individually and only looking in requirements. It's a broader, simpler search that will find used_vars in more complex template logic.